api/src/pcapi/routes/pro/collective_offers.py

```diff
@@ -324,6 +330,15 @@ def patch_all_collective_offers_active_status(
 def patch_collective_offers_active_status(
     body: collective_offers_serialize.PatchCollectiveOfferActiveStatusBodyModel,
 ) -> None:
+    if body.is_active == True:
+        for offer_id in body.ids:
+            offer = educational_repository.get_collective_offer_by_id(offer_id)
+            offerer_id = offer.venue.managingOffererId
+            try:
+                offerers_api.can_offerer_create_educational_offer(offerer_id)
```

Comment:
On va faire une requête HTTP vers Adage par offre, alors qu'il est très probable que toutes les offres sont liées à la même structure. On pourrait plutôt d'abord récupérer l'ensemble des structures distinctes, puis itérer dessus : 

```python
    offerer_ids = set()
    for offer_id in body.ids:
        offerer_ids.add(...)
    for offerer_id in offerer_ids:
        can_offerer_create_educational_offer(offerer_id)
        ...
```